openstack/identity/v3/credentials/requests.go

```diff
@@ -0,0 +1,123 @@
+package credentials
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// ListOptsBuilder allows extensions to add additional parameters to
+// the List request
+type ListOptsBuilder interface {
+	ToCredentialListQuery() (string, error)
+}
+
+// ListOpts provides options to filter the List results.
+type ListOpts struct {
+	// UserID filters the response by a credential user_id
+	UserID string `json:"user_id"`
+}
+
+// ToCredentialListQuery formats a ListOpts into a query string.
+func (opts ListOpts) ToCredentialListQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	return q.String(), err
+}
+
+// List enumerates the Credentials to which the current token has access.
+func List(client *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pager {
+	url := listURL(client)
+	if opts != nil {
+		query, err := opts.ToCredentialListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+	return pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
+		return CredentialPage{pagination.LinkedPageBase{PageResult: r}}
+	})
+}
+
+// Get retrieves details on a single user, by ID.
+func Get(client *gophercloud.ServiceClient, id string) (r GetResult) {
+	_, r.Err = client.Get(getURL(client, id), &r.Body, nil)
+	return
+}
+
+// CreateOptsBuilder allows extensions to add additional parameters to
+// the Create request.
+type CreateOptsBuilder interface {
+	ToCredentialCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts provides options used to create a credential.
+type CreateOpts struct {
+	// Serialized blob containing the credentials
+	Blob string `json:"blob" required:"true"`
+	// ID of the project.
+	ProjectID string `json:"project_id" required:"true"`
```

Comment:
`project_id` is not required, when the type is not `ec2`. https://github.com/openstack/keystone/blob/00663f93b7e9513602d73dcd0aba7ad1cc1feefc/keystone/credential/schema.py#L44